pystac/collection.py

```diff
@@ -151,16 +151,10 @@ class TemporalExtent:
         Datetimes are required to be in UTC.
     """
 
-    def __init__(
-        self, intervals: Union[List[List[Optional[Datetime]]], List[Optional[Datetime]]]
-    ):
-        # A common mistake is to pass in a single interval instead of a
-        # list of intervals. Account for this by transforming the input
-        # in that case.
-        if isinstance(intervals, list) and isinstance(intervals[0], Datetime):
-            self.intervals = [cast(List[Optional[Datetime]], intervals)]
-        else:
-            self.intervals = cast(List[List[Optional[Datetime]]], intervals)
+    def __init__(self, intervals: List[List[Optional[Datetime]]]):
```

Comment:
This changes behavior; I'd prefer to keep this PR to the type changes. If we want to remove this nicety, which was put in place based on user requests, we can discuss in a follow up PR or issue.